infrastructures/infrastructure-azure/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AzureInfrastructure.java

```diff
@@ -408,28 +418,49 @@ private String generateDefaultRMHostname() {
         }
     }
 
-    private String generateScriptFromInstanceId(String instanceId) {
+    private String generateScriptFromInstanceId(String instanceId, String osType) {
+        // if the script didn't change compared to the previous update,
+        // then the script will not be run by the Azure provider.
+        // Moreover, the 'forceUpdateTag' is not available in the Azure
+        // Java SDK. So in order to make sure that the script slightly
+        // changes, we make the scripts begin with a command that prints
+        // a new ID each time a script needs to be run after the first
+        // executed script.
+
+        UUID scriptExecutionId = UUID.randomUUID();
+        String uniqueCommandPart = "echo script-ID" + "; " + "echo " + scriptExecutionId + "; ";
+
         String startNodeCommand = generateStartNodeCommand(instanceId);
-        if (System.getProperty("os.name").contains("Windows")) {
-            return this.downloadCommand + "; Start-Process -NoNewWindow " + startNodeCommand;
+        if (osType.equals("windows")) {
+            String[] splittedStartNodeCommand = startNodeCommand.split(" ");
+            StringBuilder downloadCommand = new StringBuilder("powershell -command \"" + uniqueCommandPart +
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "downloadCommand" which hides the field declared at line 198. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)

Reply:
Done in 1a0a30e01f88f1d6f150678762c156731fd457ee